webapp/src/ts/modules/tasks/tasks.component.ts

```diff
@@ -142,7 +155,29 @@ export class TasksComponent implements OnInit, OnDestroy {
       .then(taskDocs => {
         this.hasTasks = taskDocs.length > 0;
         this.loading = false;
-        this.tasksActions.setTasksList(this.hydrateEmissions(taskDocs));
+        this.hydrateEmissions(taskDocs)
+          .then((tasks) => {
+            // get lineages for all tasks
+            this.getLineagesFromTaskDocs(tasks)
+              .then((subjects) => {
+                const deepCopy = obj => JSON.parse(JSON.stringify(obj));
+                const lineagedTasks = deepCopy(tasks);
+                lineagedTasks?.forEach((task) => {
```

Comment:
This logic is hard to read and it's adding a lot of nested code blocks. This can be an indicator of code smell and that it can be structured better. We try to keep code as flat as possible.